Base/QTCore/qSlicerCoreCommandOptions.cxx

```diff
@@ -90,15 +92,30 @@ bool qSlicerCoreCommandOptions::parse(const QStringList& arguments)
     return false;
     }
 
-  // If first unparsed argument is python script, enable 'shebang' mode
+  // Enable running Python script .py file when it is the first argument ('shebang' mode).
+  // Also accept if there is a -I argument before the .py file, because this a commonly used
+  // command-line switch of Python.exe to "isolate Python from the user's environment".
+  // For example, VS Code runs Slicer using these arguments to query its Python interpreter version:
+  // .../Slicer.exe -I ...\.vscode\extensions\ms-python.python-2023.6.1\pythonFiles\get_output_via_markers.py
+  //   ...\.vscode\extensions\ms-python.python-2023.6.1\pythonFiles\interpreterInfo.py
   QStringList unparsedArguments = this->unparsedArguments();
-  if (unparsedArguments.size() > 0 && unparsedArguments.at(0).endsWith(".py"))
+  if ((unparsedArguments.size() > 0 && unparsedArguments.at(0).endsWith(".py"))
+    || (unparsedArguments.size() > 1 && unparsedArguments.at(0) == "-I" && unparsedArguments.at(1).endsWith(".py")))
     {
     if(!this->pythonScript().isEmpty())
       {
       qWarning() << "Ignore script specified using '--python-script'";
       }
-    this->setExtraPythonScript(unparsedArguments.at(0));
+    if (unparsedArguments.at(0) == "-I")
+      {
+      this->setExtraPythonScriptProcessedArgumentsCount(2);
+      this->setExtraPythonScript(unparsedArguments.at(1));
+      }
+    else
+      {
+      this->setExtraPythonScriptProcessedArgumentsCount(1);
+      this->setExtraPythonScript(unparsedArguments.at(0));
+      }
     this->setRunPythonAndExit(true);
     }
 
```

Comment:
Next should be replaced like this:

```diff
-  if (!d->ParsedArgs.value("c").toString().isEmpty())
+  else if (!d->ParsedArgs.value("c").toString().isEmpty())
```

Reply:
I'm not sure what you mean. Do you suggest some code changes or just commented on how we could interpret some more extra Python script arguments in the future?

Reply:
I suggested a simpler implementation for doing the same thing. Including the suggestion above and this one should do it.

Limitation of the GitHub interface prevented me from doing the suggestion in one go.